jerry-core/ecma/base/ecma-helpers-values-collection.cpp

```diff
@@ -38,7 +38,7 @@ ecma_new_values_collection (const ecma_value_t values_buffer[], /**< ecma-values
 {
   JERRY_ASSERT (values_buffer != NULL || values_number == 0);
 
-  const size_t values_in_chunk = sizeof (ecma_collection_chunk_t::data) / sizeof (ecma_value_t);
+  const size_t values_in_chunk = sizeof (((ecma_collection_chunk_t *)0)->data) / sizeof (ecma_value_t);
```

Comment:
Please do not use a hack for getting the size. Instead, you should use macros.

```
 lit_utf8_byte_t data[ sizeof (uint64_t) - sizeof (mem_cpointer_t) ];
```

For example:
```
#define ECMA_COLLECTION_DATA_COUNT (sizeof (uint64_t) - sizeof (mem_cpointer_t))
#define ECMA_COLLECTION_DATA_SIZE (sizeof (lit_utf8_byte_t) * ECMA_COLLECTION_DATA_COUNT)

lit_utf8_byte_t data[ ECMA_COLLECTION_DATA_COUNT ];
```
```
const size_t values_in_chunk = ECMA_COLLECTION_DATA_SIZE;
```

As I wrote (https://github.com/Samsung/jerryscript/pull/806#issuecomment-179133239) I'm already eliminating the C++ features. So the current workflow not the best because it results work duplication. The example that I show is my result for this problem.